code-experiments/src/transform_vars_blockrotation_helpers.c

```diff
@@ -90,27 +90,30 @@ static void coco_compute_blockrotation(double **B, long seed, size_t n, size_t *
   double **current_block;
   size_t i, j, k; /* Loop over pairs of column vectors. */
   size_t idx_block, current_blocksize,cumsum_prev_block_sizes, sum_block_sizes;
-  size_t nb_entries;
-  coco_random_state_t *rng = coco_random_new((uint32_t) seed);
-  
+  size_t nb_entries, current_nb_entries;
```

Comment:
Better:
```C
size_t nb_entries;
size_t current_nb_entries;
```
because two declarations in one line are, IIRC, not allowed in the C89 ("ANSI C") standard.

Reply:
It's bad, but I don't think C89 is that bad. The restriction is that variables can only be defined at the beginning of a scope. Declaring several variables at once is IMHO reasonable coding style.